pymeasure/instruments/thermotron/thermotron3800.py

```diff
@@ -0,0 +1,129 @@
+    """ Represents the Thermotron 3800 Oven.
+    For now, this driver only supports using Control Channel 1.
+    """
+
+    def __init__(self, resourceName, **kwargs):
+        super().__init__(
+            resourceName,
+            "Thermotron 3800",
+            **kwargs
+        )
+
+    id = Instrument.measurement(
+        "IDEN?", """ Reads the instrument identification 
+        
+        :return: String
+        """
+    )
+
+    temperature = Instrument.measurement(
+        "PVAR1?", """ Reads the current temperature of the oven 
+        via built in thermocouple
+        
+        :return: float
+        """
+    )
+
+    mode = Instrument.measurement(
+        "MODE??", """ Gets the operating mode of the oven.
+        
+        :return: Tuple(String, int)
+        """,
+        get_process=lambda mode: Thermotron3800.__translate_mode(mode)
+    )
+    
+    setpoint = Instrument.control(
+        "SETP1?", "SETP1,%g",
+        """ A floating point property that controls the setpoint 
+        of the oven in Celsius. This property can be set. 
+        
+        Insert wait time after this command. This wait time should be >1000ms for consistent results.
+        Failing to wait for an adequate time may cause errors in subsequent oven commands.
+        
+        :return: None
+        """,
+        validator=strict_range,
+        values=[-55, 150]
+    )
+
+    def run(self):
+        '''
+        Insert wait time after this command. This wait time should be >1000ms for consistent results.
+        Failing to wait for an adequate time may cause errors in subsequent oven commands.
+        :return: None
+        '''
+        self.write("RUNM")
+
+    def stop(self):
+        '''
+        Insert wait time after this command. This wait time should be >1000ms for consistent results.
+        Failing to wait for an adequate time may cause errors in subsequent oven commands.
+        :return: None
+        '''
+        self.write("STOP")
+
+    def initalize_oven(self):
+        '''
+        Please wait 3 seconds after calling initialize_oven before running
+        any other oven commands (per manufacterer's instructions).
+
+        :return: None
+        '''
+        self.write("INIT")
+
+    @staticmethod
+    def __translate_mode(mode_coded_integer):
+        '''
+        Bit 0 = Program mode
+        Bit 1 = Edit mode (controller in stop mode)
+        Bit 2 = View program mode
+        Bit 3 = Edit mode (controller in hold mode)
+        Bit 4 = Manual mode
+        Bit 5 = Delayed start mode
+        Bit 6 = Unused
+        Bit 7 = Calibration mode
+        '''
+        map = {
+            1: "Program mode",
+            2: "Edit mode (controller in stop mode)",
+            4: "View program mode",
+            8: "Edit mode (controller in hold mode)",
+            16: "Manual mode",
+            32: "Delayed start mode",
+            64: "Unused (Error)",
+            128: "Calibration mode"
+        }
+
```

Comment:
Actually, I don't think a combined mode is possible, so I'll remove the comment about the "combined mode". 

Regarding the Python Enum, do you mean something like the following?:
```Python
class Mode(Enum):
    field1 = "string"
    ...
```
I chose the map because I could easily map the integer I get back from the oven to a string. I'm not sure how to do that with the Enum. Can you please help explain what you mean?

Reply:
Something like this?
``` 
class Mode(IntFlag):
    PROGRAM_MODE = 1
    EDIT_MODE = 2
...  #(caps are only convention)
```
See [docs](https://docs.python.org/3/library/enum.html#intflag). Depending on your needs, an `IntEnum` could also be sufficient (if you don't need bitwise combinations - maybe this is the "combined mode" you're referring to?)

Reply:
The enum member you get back is also readable in the console, the name should be sufficiently understandable without having separate strings, and you only need to return one object instead of a tuple from `self.mode`.

Reply:
Hi bilderbuchi, thanks so much for the suggestions! I've added a mode enum to the class, and also added support for that "combined mode" that I mentioned earlier. I used the IntFlag enum for this. I'm not actually sure if it is possible for the oven to be in multiple modes at the same time (which is what I meant by combined mode), but if it ever does happen, the __translate_mode function will now return an enum that has all of the modes OR'd together. For example, it could return:
```Python
Thermotron3800Mode.CALIBRATION_MODE|MANUAL_MODE
```
if the oven is in calibration mode and manual mode. If it is in an unknown mode (mode = 0 or greater than 255), then __translate_mode will return 
```Python
Thermotron3800Mode.UNKNOWN_MODE
```

Please let me know if you think any of this behavior could be tweaked to be better, or if it could be implemented in a better way. I appreciate your feedback. Thanks!

Reply:
[Link to commit](https://github.com/pymeasure/pymeasure/pull/606/commits/7152c11cb1019ded906a939856dfc11831c0d7fe)

Sorry about the linter issues. I believe I have fixed them in future commits.

Reply:
I don't have time right now to review that in detail, but iirc `IntFlag` should be able to do the "or" combination as-is, without any custom code needed. Did you try that before?

E.g., with the `Mode` snippet from above:
```
class Mode(IntFlag):
    PROGRAM_MODE = 1
    EDIT_MODE = 2

Mode(3)  # prints <Mode.EDIT_MODE|PROGRAM_MODE: 3>
```

Reply:
Addressed by committ 86b2d2a99c9743ebc422ce616ec431533857b7ef as suggested by @bilderbuchi